File: articles/sensitivity-analysis-for-algorithms-and-models/julia/sensitivity_analysis.jl
# Minimal Julia example for sensitivity-analysis parameter sweeps.
model(demand, capacity, failure, adaptation) = clamp(0.5 + 0.3demand + 0.25failure - 0.2capacity - 0.15adaptation, 0.0, 1.0)
base = Dict(:demand => 0.45, :capacity => 0.35, :failure => 0.25, :adaptation => 0.30)
println("baseline_risk=", model(base[:demand], base[:capacity], base[:failure], base[:adaptation]))

File: articles/sensitivity-analysis-for-algorithms-and-models/c/sensitivity_analysis.c
#include <stdio.h>

double clamp(double x) { return x < 0 ? 0 : (x > 1 ? 1 : x); }
double model(double demand, double capacity, double failure, double adaptation) {
    return clamp(0.5 + 0.30*demand + 0.25*failure - 0.20*capacity - 0.15*adaptation);
}
int main(void) {
    printf("baseline_risk=%.6f\n", model(0.45, 0.35, 0.25, 0.30));
    return 0;
}
